File: JLBannerViewDemo/JLBannerViewDemo/CollectionViewController.m
//
//  CollectionViewController.m
//  JLBannerViewDemo
//
//  Created by Jangsy7883 on 2017. 4. 26..
//  Copyright © 2017년 Studio. All rights reserved.
//

#import "CollectionViewController.h"
#import "CastBannerCell.h"



@interface CollectionViewController () <JLBannerViewDelegate, JLBannerViewDataSource>

@end

@implementation CollectionViewController

static NSString * const reuseIdentifier = @"BannerCell";

- (void)viewDidLoad {
    [super viewDidLoad];

    [self.collectionView registerClass:[CastBannerCell class] forCellWithReuseIdentifier:reuseIdentifier];
}

#pragma mark <UICollectionViewDataSource>

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return 1;
}

- (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout sizeForItemAtIndexPath:(NSIndexPath *)indexPath {
    return CGSizeMake(collectionView.bounds.size.width,
                      collectionView.bounds.size.width*0.389);
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    CastBannerCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:reuseIdentifier forIndexPath:indexPath];
    cell.backgroundColor = [[UIColor blueColor] colorWithAlphaComponent:0.7];
    
    cell.bannerView.delegate = self;
    cell.bannerView.dataSource = self;
    cell.bannerView.autoScrolling = YES;

    return cell;
}

#pragma mark - JLBannerViewDataSource

- (NSInteger)numberOfItemsInBannerView:(JLBannerView *)bannerView {
    return 10;
}

- (UIView *)bannerView:(JLBannerView *)bannerView reusableItemView:(UIView *)reusableItemView forItemAtIndex:(NSInteger)index {
    
    UILabel *label = (UILabel *)reusableItemView;
    if (label == nil || ![label isKindOfClass:[UILabel class]]) {
        label = [[UILabel alloc] init];
        label.textAlignment = NSTextAlignmentCenter;
        label.font = [UIFont systemFontOfSize:35 weight:UIFontWeightMedium];
        label.textColor = [UIColor whiteColor];
    }
    
    label.text = [NSString stringWithFormat:@"%zd",index];
    return label;
}

#pragma mark - JLBannerViewDelegate

- (void)bannerView:(JLBannerView *)bannerView didSelectItemAtIndex:(NSInteger)index {
//    NSLog(@"selected %zd",index);
}
- (void)bannerView:(JLBannerView *)bannerView didScrollToIndex:(NSInteger)index {
//    NSLog(@"scroll %zd",index);
}

@end

File: JLBannerViewDemo/JLBannerViewDemo/CollectionViewController.swift
//
//  CollectionViewController.swift
//  JLBannerViewDemo
//
//  Created by Woody on 2017. 5. 25..
//  Copyright © 2017년 Woody. All rights reserved.
//

import UIKit

private let reuseIdentifier = "BannerCell"

class CollectionViewController: UICollectionViewController, JLBannerViewDelegate, JLBannerViewDataSource,UICollectionViewDelegateFlowLayout {

    // MARK: - View lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.collectionView!.register(CastBannerCell.self, forCellWithReuseIdentifier: reuseIdentifier)
    }
    
    // MARK: - UICollectionViewDataSource

    override func numberOfSections(in collectionView: UICollectionView) -> Int {
        return 1
    }

    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 1
    }

    // MARK: - UICollectionViewDelegate
    
    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! CastBannerCell
        cell.backgroundColor = UIColor.blue.withAlphaComponent(0.7)
        
        cell.bannerView?.delegate = self;
        cell.bannerView?.dataSource = self;
        cell.bannerView?.isAutoScrolling = true;
        return cell
    }

    // MARK: - UICollectionViewDelegateFlowLayout
    
    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
        return CGSize(width: collectionView.bounds.size.width, height: collectionView.bounds.size.width*0.389)
    }

    // MARK: - JLBannerViewDataSource
    
    func numberOfItems(in bannerView: JLBannerView!) -> Int {
        return 10;
    }
    
    func bannerView(_ bannerView: JLBannerView!, reusableItemView: UIView!, forItemAt index: Int) -> UIView! {
        var label: UILabel!
        if reusableItemView is UILabel {
            label = reusableItemView as! UILabel
        }
        else {
            label = UILabel()
            label.textAlignment = .center
            label.font = UIFont.systemFont(ofSize: 35, weight: UIFontWeightMedium)
            label.textColor = UIColor.white
        }
        
        label.text = String(index)
        return label
    }
    
    // MARK: - JLBannerViewDelegate
    
    func bannerView(_ bannerView: JLBannerView!, didScrollTo index: Int) {
        print("\(#function) \(index)")
    }
    
    func bannerView(_ bannerView: JLBannerView!, didSelectItemAt index: Int) {
        print("\(#function) \(index)")
    }
}
